src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs

```diff
@@ -35,5 +35,31 @@ public static class TensorflowCatalog
         /// </example>
         public static TensorFlowModel LoadTensorFlowModel(this ModelOperationsCatalog catalog, string modelLocation)
             => TensorFlowUtils.LoadTensorFlowModel(CatalogUtils.GetEnvironment(catalog), modelLocation);
+
+        /// <summary>
+        /// Load TensorFlow model into memory. This is the convenience method that allows the model to be loaded once and subsequently use it for querying schema and creation of
+        /// <see cref="TensorFlowEstimator"/> using <see cref="TensorFlowModel.ScoreTensorFlowModel(string, string, bool)"/>.
+        /// usage of this API requires additional NuGet dependencies on TensorFlow redist, see linked document for more information.
+        /// <see cref="TensorFlowModel"/> also holds references to unmanaged resources that need to be freed either with an explicit
+        /// call to Dispose() or implicitly by declaring the variable with the "using" syntax/>
+        ///
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!include[io](~/../docs/samples/docs/api-reference/tensorflow-usage.md)]
+        /// ]]>
+        /// </format>
+        /// </summary>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="modelLocation">Location of the TensorFlow model.</param>
+        /// <param name="treatOutputAsBatched">If the first dimension of the output is unknown, should it be treated as batched or not.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[LoadTensorFlowModel](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/TextClassification.cs)]
+        /// ]]>
+        /// </format>
+        /// </example>
+        public static TensorFlowModel LoadTensorFlowModel(this ModelOperationsCatalog catalog, string modelLocation, bool treatOutputAsBatched)
```

Comment:
Do we think in the future we may get more options than `treatOutputAsBatched`? I am asking to know if it will be better to have this as enum value instead of boolean?

Reply:
That is a good question that I don't know the answer too. From my understanding, this is the first change to these API's since there were first added, so my guess is that if things are added it wont be frequently, but I don't know if more will be added or not. 

I chose a bool this time because the other parameter dealing with batched input was a bool. I am happy to change it or discuss it more if you would like. I am not super familiar with the .NET guidelines, so it would be good for me to learn more.

Reply:
Here is the guidlines for such case: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/parameter-design#choosing-between-enum-and-boolean-parameters

I would prefer using enum here for future flexibility.

Reply:
So for the specific case of treating the output as batched or not, it will only ever be a yes or no answer. We can't do it half and half. Are the guidelines meaning that we want to be able to combine flags?

Reply:
It is not about combining flags. let's say you'll define the enum as 

```C#
public enum TensorflowCatalogOptions // Of course, you can choose better name
{
    None, 
    Batched
}
```

the API will be 

```C#
public static TensorFlowModel LoadTensorFlowModel(this ModelOperationsCatalog catalog, string modelLocation, TensorflowCatalogOptions options)
```

for any other method take this as defaulted parameter will do 


```C#
internal TensorFlowModel(IHostEnvironment env, Session session, string modelLocation, TensorflowCatalogOptions options = TensorflowCatalogOptions.Batched)
```

and in the implementation, you can test against TensorflowCatalogOptions.Batched value. 


Imagin in the future you will get a new case you want to support, say Filtered. which you can just add to the enum and handle it in the implmentation.

```C#
public enum TensorflowCatalogOptions // Of course, you can choose better name
{
    None, 
    Batched, 
    Filtered
}
```

It is up to you too if you want to make this enum as Flags enum which allow passing multiple values to this enum.

```C#
[Flags]
public enum TensorflowCatalogOptions // Of course, you can choose better name
{
    None = 0, 
    Batched = 1, 
    Filtered = 2
}
```

And that can allow passing multiple values if needed, like `TensorflowCatalogOptions.Batched | TensorflowCatalogOptions.Filtered`. And in the implmentation you can check specific case like `if (options & TensorflowCatalogOptions.Batched) != 0`.

In short, you need to think if this is possible in the future you can support more than options? and if it is possible multiple options can be supported together.

Reply:
So thinking about it, unless TensorFlow changes how they deal with their dimensions there won't be anything else that this could be. They only use ints to represent the dimensions so they are limited in how they express them, and essentially its that negative numbers mean unknown size and positive numbers mean known size. They would have to completely change how they represented the dimensions for that to change, which would then require a complete rewrite on our side to handle it if they did that. So I think we are safe to assume it will only be 2 options here, so I think we are safe to keep it as a boolean.

Reply:
You know better than me here :-) so I'll let you decide about it.